auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -162,12 +162,22 @@ private void populateIdpList(List<IdpConfig> providerConfigs,
                     this, R.string.fui_progress_dialog_loading) {
                 @Override
                 protected void onSuccess(@NonNull IdpResponse response) {
-                    handler.startSignIn(response);
+                    handleResponse(response);
                 }
 
                 @Override
                 protected void onFailure(@NonNull Exception e) {
-                    handler.startSignIn(IdpResponse.from(e));
+                    handleResponse(IdpResponse.from(e));
+                }
+
+                private void handleResponse(@NonNull IdpResponse response) {
+                    if (!response.isSuccessful()
```

Comment:
Can we simplify this if statement?  It's kinda melting my brain. 

```java
if (AuthUI.SOCIAL_PROVIDERS.contains(response.getProviderType())) {
  handler.startSignIn(response)
} else if (response.isSuccessful()) {
  finish(RESULT_OK, response.toIntent();
} else {
  finish(RESULT_CANCELED, response.toIntent());
}
```

Reply:
Unfortunately not, your thing will crash. The only reason I have the successful check is because the providerId will be null if it's a failure. Other ideas? (I agree that the statement is confusing, but that nullability is biting us in the behind.)